File: src/main/java/workflow/process/events/FileEventProducer.java
package workflow.process.events;

import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.stereotype.Service;


@Service
@RequiredArgsConstructor
public class FileEventProducer {

    private final RabbitTemplate rabbitTemplate;

    public void produceFileEvent(@NotNull final String fileUUID) {
        final FileProcessEvent event = new FileProcessEvent();
        event.setFileUUID(fileUUID);
        rabbitTemplate.convertAndSend("file-events", event);
    }
}

File: src/main/kotlin/workflow/process/events/FileEventProducer.kt
package workflow.process.events

import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.stereotype.Service


@Service
/**
 * Produces file events to rabbitmq
 */
class FileEventProducer(private val rabbitTemplate: RabbitTemplate) {

    fun produceFileEvent(fileName: String) {
        rabbitTemplate.convertAndSend("file-events", FileProcessEvent(fileName))
    }
}